x-pack/test/alerting_api_integration/security_and_spaces/tests/alerting/disable.ts

```diff
@@ -84,6 +84,62 @@ export default function createDisableAlertTests({ getService }: FtrProviderConte
           }
         });
 
+        it('should still be able to disable alert when AAD is broken', async () => {
+          const { body: createdAlert } = await supertest
+            .post(`${getUrlPrefix(space.id)}/api/alert`)
+            .set('kbn-xsrf', 'foo')
+            .send(getTestAlertData({ enabled: true }))
+            .expect(200);
+          objectRemover.add(space.id, createdAlert.id, 'alert');
+
+          await supertest
+            .put(`${getUrlPrefix(space.id)}/api/saved_objects/alert/${createdAlert.id}`)
+            .set('kbn-xsrf', 'foo')
+            .send({
+              attributes: {
+                name: 'bar',
+              },
+            })
+            .expect(200);
+
+          const response = await alertUtils.getDisableRequest(createdAlert.id);
+
+          switch (scenario.id) {
+            case 'no_kibana_privileges at space1':
+            case 'space_1_all at space2':
+            case 'global_read at space1':
+              expect(response.statusCode).to.eql(404);
+              expect(response.body).to.eql({
+                statusCode: 404,
+                error: 'Not Found',
+                message: 'Not Found',
+              });
+              // Ensure task still exists
+              await getScheduledTask(createdAlert.scheduledTaskId);
+              break;
+            case 'superuser at space1':
+            case 'space_1_all at space1':
+              expect(response.statusCode).to.eql(204);
+              expect(response.body).to.eql('');
+              try {
+                await getScheduledTask(createdAlert.scheduledTaskId);
+                throw new Error('Should have removed scheduled task');
+              } catch (e) {
+                expect(e.status).to.eql(404);
+              }
+              // Ensure AAD isn't broken
+              await checkAAD({
```

Comment:
I guess AAD is now "fixed" (it was "broken" just above), because the `enable()` call "fixed" it.  Interesting.  That seems fine for alerts, I wonder if it would end up being a security issue with actions, when we get to fixing it's related encryption key issues.

Reply:
Yeah, this scenario we're lucky. For actions, I believe all the secrets have to be passed in for any sort of update to the SO. We probably won't have this issue of loading the decrypted object beforehand in the actions client, I think.